src/vs/code/electron-main/app.ts

```diff
@@ -163,28 +163,88 @@ export class CodeApplication extends Disposable {
 		// !!! DO NOT CHANGE without consulting the documentation !!!
 		//
 
+		const isUrlFromWindow = (requestingUrl?: string | undefined) => requestingUrl?.startsWith(`${Schemas.vscodeFileResource}://${VSCODE_AUTHORITY}`);
 		const isUrlFromWebview = (requestingUrl: string | undefined) => requestingUrl?.startsWith(`${Schemas.vscodeWebview}://`);
 
 		const allowedPermissionsInWebview = new Set([
 			'clipboard-read',
 			'clipboard-sanitized-write',
 		]);
 
+		const allowedPermissionsInCore = new Set([
+			'media'
+		]);
+
 		session.defaultSession.setPermissionRequestHandler((_webContents, permission, callback, details) => {
 			if (isUrlFromWebview(details.requestingUrl)) {
 				return callback(allowedPermissionsInWebview.has(permission));
 			}
 
+			if (isUrlFromWindow(details.requestingUrl)) {
+				return callback(allowedPermissionsInCore.has(permission));
+			}
+
 			return callback(false);
 		});
 
 		session.defaultSession.setPermissionCheckHandler((_webContents, permission, _origin, details) => {
 			if (isUrlFromWebview(details.requestingUrl)) {
 				return allowedPermissionsInWebview.has(permission);
 			}
-
+			if (isUrlFromWindow(details.requestingUrl)) {
+				return allowedPermissionsInCore.has(permission);
+			}
 			return false;
 		});
+		session.defaultSession.setDisplayMediaRequestHandler(async (request, callback) => {
```

Comment:
I wonder if there is something that Electron can do here to make this easier, or does every app have to implement this code to figure out the active window screen?

Reply:
The handler is meant to act as an entry point for applications to create the screen picker and the user would go onto select the screen that is meant to be captured.

The usage here is specializing it further to autoselect the active window without a user gesture. I still feel the capture api is a bit heavy for the use case in this PR. Please correct me on why the alternative does not work.